Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModuleEditor.cs

```diff
@@ -2,14 +2,25 @@
 using System;
 using System.Linq;
 using UnityEditor;
+using UnityEditor.EventSystems;
+using UnityEngine.InputSystem.Editor;
 
 ////TODO: add button to automatically set up gamepad mouse cursor support
 
 namespace UnityEngine.InputSystem.UI.Editor
 {
     [CustomEditor(typeof(InputSystemUIInputModule))]
+    [InitializeOnLoad]
     internal class InputSystemUIInputModuleEditor : UnityEditor.Editor
     {
+        static InputSystemUIInputModuleEditor()
+        {
+#if UNITY_6000_0_OR_NEWER
+            if (EditorPlayerSettingHelpers.newSystemBackendsEnabled && !EditorPlayerSettingHelpers.oldSystemBackendsEnabled)
+                InputModuleComponentFactory.SetInputModuleComponentOverride(go => go.AddComponent<InputSystemUIInputModule>());
```

Comment:
I did some more testing on uGUI. This will have to be replaced by
```
ObjectFactory.AddComponent<InputSystemUIInputModule>(go)
```
, otherwise we have a weird bug with Undo. See gif below:

![Unity_9y36X5MqV9](https://github.com/Unity-Technologies/InputSystem/assets/35855598/eda37f8a-6f78-463a-991c-06fc717b6d7a)

Reply:
Good catch ⚡ Replaced it